### PR TITLE
Fix type information when using dynamic projection and generic repository

### LINKED ISSUE #148: version 3.2.1 is losing jpa parameters type when using dynamic projections and generics
### What steps will reproduce the problem ?
1. create a super repository interface using generics
```
@NoRepositoryBean
public interface EntityGraphBaseRepository<T extends EntityId<I>, I extends Serializable> extends EntityGraphJpaRepository<T, I> {
    <X> Optional<X> findById(I id, Class<X> clazz);
}
```

2. create your repository interface extending the your repository```

@Repository
public interface MyEntityRepository extends EntityGraphBaseRepository<MyEntity, Long> {}
```

3. then try to use the method
<X> Optional<X> findById(I id, Class<X> clazz);

### What is the expected output ?
a projection of type class<X> should be returned

### What happens instead ?
a exception is thrown
Caused by: org.hibernate.query.SemanticException: Cannot compare left expression of type 'java.lang.Long' with right expression of type 'java.io.Serializable'


### Environment
 - Spring Data JPA version (not the Spring Boot one): 
 3.2.1
 
 - ORM with version: 
 hibernate 6.4.1
 
 - spring-data-jpa-entity-graph version: 
 3.2.1

### Link to an automated test demonstrating the problem
https://filetransfer.io/data-package/wF7qTO47#link

### Additional context
Spring boot updated to version 3.2.1 and hibernate 6.4.1, the method org.springframework.data.repository.query.QueryMethod#createParameters(java.lang.reflect.Method) was deprecated in favor of 
org.springframework.data.repository.query.QueryMethod#createParameters(org.springframework.data.repository.query.ParametersSource). The entity graph library is still using the deprecated version, then some metadata information is missed and hibernate type check failing.

### PR DESCRIPTION
**Pull request by leassis**
fix: the type information when using generic repository along with dynamic projections.

closes #148

### CODE REVIEW

**Comment by reda-alaoui** [comment]
Hello @leassis ,
Please fix the formatting issue.

**Comment by leassis** [comment]
Do you have a standard formatter?

**Comment by reda-alaoui** [comment]
Yes, [this one](https://github.com/Cosium/git-code-format-maven-plugin)

You can run [this](https://github.com/Cosium/git-code-format-maven-plugin?tab=readme-ov-file#manual-code-formatting) to format your code.

**Comment by leassis** [comment]
done!

**Review by reda-alaoui — CHANGES_REQUESTED** [review]
Everything looks fine to me.

But this introduces a breaking change regarding the support of Spring Data JPA. Therefore, the README's Compatibility matrix (https://github.com/Cosium/spring-data-jpa-entity-graph?tab=readme-ov-file#compatibility-matrix) should be changed to show that:
- Spring Data JPA `3.2.1+` is supported by Spring Data JPA Entity Graph `3.2.2+`
- Spring Data JPA `3.2.0` is supported by Spring Data JPA Entity Graph `3.2.0-3.2.1`

File: README.md [hunk 1]
```diff
@@ -313,6 +313,8 @@ You can play with https://github.com/Cosium/spring-data-jpa-entity-graph-sample
...
+| 3.2.1+                                                                        | [![Maven Central 3.2.2+](https://img.shields.io/maven-central/v/com.cosium.spring.data/spring-data-jpa-entity-graph/3.2.2.svg)](https://search.maven.org/#search%7Cgav%7C1%7Cg%3A%22com.cosium.spring.data%22%20AND%20a%3A%22spring-data-jpa-entity-graph%22)                                                                                                                                                                                                                                                             |
+| 3.2.0                                                                         | [![Maven Central 3.2.0](https://img.shields.io/maven-central/v/com.cosium.spring.data/spring-data-jpa-entity-graph/3.2.0.svg)](https://search.maven.org/#search%7Cgav%7C1%7Cg%3A%22com.cosium.spring.data%22%20AND%20a%3A%22spring-data-jpa-entity-graph%22) [![Maven Central 3.2.0](https://img.shields.io/maven-central/v/com.cosium.spring.data/spring-data-jpa-entity-graph/3.2.1.svg)](https://search.maven.org/#search%7Cgav%7C1%7Cg%3A%22com.cosium.spring.data%22%20AND%20a%3A%22spring-data-jpa-entity-graph%22) |
```
Comment (reda-alaoui) [inline README.md:317]:
```suggestion
| 3.2.0                                                                         | [![Maven Central 3.2.1](https://img.shields.io/maven-central/v/com.cosium.spring.data/spring-data-jpa-entity-graph/3.2.1.svg)](https://search.maven.org/#search%7Cgav%7C1%7Cg%3A%22com.cosium.spring.data%22%20AND%20a%3A%22spring-data-jpa-entity-graph%22) |
```

File: README.md [hunk 2]
```diff
@@ -313,6 +313,8 @@ You can play with https://github.com/Cosium/spring-data-jpa-entity-graph-sample
...
 | 3.2.x                                                                         | [![Maven Central 3.2.x](https://img.shields.io/maven-central/v/com.cosium.spring.data/spring-data-jpa-entity-graph/3.2.svg)](https://search.maven.org/#search%7Cgav%7C1%7Cg%3A%22com.cosium.spring.data%22%20AND%20a%3A%22spring-data-jpa-entity-graph%22)   |
```
Comment (reda-alaoui) [inline README.md:318]:
```suggestion
```

**Review by reda-alaoui — CHANGES_REQUESTED** [review]

**Review by reda-alaoui — APPROVED** [review]

**Comment by reda-alaoui** [comment]
Thank you @leassis !

**Pull request merged**